### DISCUSSION TITLE
String version and custom numbering rule

### DISCUSSION THREAD

**Discussion by anatoly-kryzhanovsky** [fluentmigrator/fluentmigrator#1301] [Ideas] [answered]
recently i use liquibase for migrations, and i like possibility of useing string as migration id (so, i can use task id in issue tracker and so i can search faster which migration i need, or vise verse - why this migrations do something).
i try to make addon to fluentmigrator (because i extremly like it!):
in my project i have json file, which has following structure:
```
{
   "Releases": [
        {
            "Id": "01.01",
            "Migrations": [
                 {"Id": ""TTT-1"},
                 {"Id": ""TTT-2"}
             ]
    ],
    "Fixtures": [
        {"Id": "FakeDataForTTT-1", "Environments": ["Development", "test"], "ApplyAfter": "TTT-1"}
    ]    
}
```

at runtime i generate inmemory assembly, which contains my migration with [Migration] attribute, where version is calculated by order in my josn file.

this work fine until we invite new developer to project. imagene fillowing case:
we have developer A and developer B.
they start to work on new feature (F1 for developer A, and F2 for developer B), each of this feature require new migration, and they implement they.
so delveoper A has migration F1 (with version 1)
and developer B has migration F2 (again with version 1)

now developer A merge changes to dev, and we have migration F1 with version 1 in dev
now developer B try to merge changes to dev, but have merge conflict, resolve it... and what is this? he heas 
migration F1 with version 1
migration F2 with version 2

but! he has already applyed version 1 on local environment, and this is migration F2.. so we have two problem - developer B missed migration F1, and migration F2 try to applied twice!

i really don't want to set explicit number of migration, so ii will be amazing to setup order (as in) and id (as string).
order use for migration ordering
id - for find which migration already applayed

**Comment by jzabroski** [comment, marked as answer]
@anatoly-kryzhanovsky I can see the merits of your use case, but things like issue tracker should go in the migration Description attribute.  The version is just a way to guaranteed the set of migrations form a totally ordered set, and so we typically name the migration using yyyyMMdd000X where 000X is a four digit number with 0 padding.  Would that work for you?  

The benefit is that you can still query the database for your specific issue tracker tickets.
